core/scripts/chaincli/handler/keeper_launch.go

```diff
@@ -0,0 +1,452 @@
+func (k *Keeper) createKeeperJob(client cmd.HTTPClient, contractAddr, nodeAddr string) error {
+	request, err := json.Marshal(web.CreateJobRequest{
+		TOML: testspecs.GenerateKeeperSpec(testspecs.KeeperSpecParams{
+			ContractAddress:          contractAddr,
+			FromAddress:              nodeAddr,
+			EvmChainID:               int(k.cfg.ChainID),
+			MinIncomingConfirmations: 1,
+		}).Toml(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal request: %s", err)
+	}
+
+	resp, err := client.Post("/v2/jobs", bytes.NewReader(request))
+	if err != nil {
+		return fmt.Errorf("failed to create keeper job: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read error response body: %s", err)
+		}
+
+		return fmt.Errorf("unable to create keeper job: '%v' [%d]", string(body), resp.StatusCode)
+	}
+
+	log.Println("Keeper job has been successfully created")
+
+	return nil
+}
+
+func (k *Keeper) launchChainlinkNode(ctx context.Context, port int) (string, func(), error) {
+	// Create docker client to launch nodes
+	dockerClient, err := client.NewClientWithOpts(client.WithAPIVersionNegotiation())
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create docker client from env: %s", err)
+	}
+
+	// Make sure everything works well
+	if _, err = dockerClient.Ping(ctx); err != nil {
+		return "", nil, fmt.Errorf("failed to ping docker server: %s", err)
+	}
+
+	// Pull DB image if needed
+	var out io.ReadCloser
+	if _, _, err = dockerClient.ImageInspectWithRaw(ctx, defaultPOSTGRESImage); err != nil {
+		log.Println("Pulling Postgres docker image...")
+		if out, err = dockerClient.ImagePull(ctx, defaultPOSTGRESImage, types.ImagePullOptions{}); err != nil {
+			return "", nil, fmt.Errorf("failed to pull Postgres image: %s", err)
+		}
+		out.Close()
+		log.Println("Postgres docker image successfully pulled!")
+	}
+
+	// Create DB container
+	dbContainerResp, err := dockerClient.ContainerCreate(ctx, &container.Config{
+		Image: defaultPOSTGRESImage,
+		Cmd:   []string{"postgres", "-c", `max_connections=1000`},
+		Env: []string{
+			"POSTGRES_USER=postgres",
+			"POSTGRES_PASSWORD=development_password",
+		},
+		ExposedPorts: nat.PortSet{"5432": struct{}{}},
+	}, nil, &network.NetworkingConfig{}, nil, "")
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create Postgres container: %s", err)
+	}
+
+	// Start container
+	if err = dockerClient.ContainerStart(ctx, dbContainerResp.ID, types.ContainerStartOptions{}); err != nil {
+		return "", nil, fmt.Errorf("failed to start DB container: %s", err)
+	}
+	log.Println("Postgres docker container successfully created and started: ", dbContainerResp.ID)
+
+	dbContainerInspect, err := dockerClient.ContainerInspect(ctx, dbContainerResp.ID)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to inspect Postgres container: %s", err)
+	}
+
+	time.Sleep(time.Second * 10)
```

Comment:
whats the sleep for?

Reply:
i think this is there do give time for the postgres container to start before starting up the chainlink node. in the future we can improve this with something that monitors and waits(ryan and i have discussed using pulumi which would understand the dependency and wait).

Reply:
pulumi is nice because it also is defined via go code. So should be able to slip it right into places like this and keep the rest of our scripts. Although it has some of its own quirks too.

Reply:
Yes, just a short term solution to wait until contaiters are up and running